index.js

```diff
@@ -1,9 +1,24 @@
 'use strict';
 const path = require('path');
+const fs = require('fs');
 const locatePath = require('locate-path');
 
 const stop = Symbol('findUp.stop');
 
+const exists = path => {
+	if (typeof path !== 'string') {
+		throw new TypeError('Expected a string');
+	}
+
+	return fs.existsSync(path);
+};
+
+const stats = path => fs.statSync(path);
+
+const isFile = path => exists(path) ? stats(path).isFile() : false;
+
+const isDirectory = path => exists(path) ? stats(path).isDirectory() : false;
```

Comment:
These functions are vulnerable to race conditions. Use the [path-type](https://github.com/sindresorhus/path-type) module to avoid it.

Reply:
I commented at https://github.com/sindresorhus/find-up/issues/37#issuecomment-489440312, maybe `locate-path` should be used for isFile and isDirectory functionality?  Not sure if this means that `locate-path` should simply be re-exported instead of these helpers or if the helpers should use `locate-path` for the implementation.

Obviously this idea is dependent on updating `locate-path` first.  I'm not really asking for this change to happen here yet, just putting the idea out there.